docs/guides/BUILDING.md

```diff
@@ -107,7 +107,21 @@ sudo apt-get install libsdl2-dev
 
 ### Installing prerequisites on macOS
 
-On macOS, install Xcode from the Mac App Store.
+On macOS, install Xcode from the Mac App Store. After installing Xcode, install
+python3.11 using brew:
+
+```sh
+    brew install python@3.11
+    python3 -m pip install --upgrade setuptools
+    python3 -m pip install --upgrade pip
+    cd /usr/local/bin
+    ln -sf python3.11 python3
+    ln -sf python3 python
+    ln -sf pip3.11 pip3
+    ln -sf pip3 pip
```

Comment:
This doesn't quite make sense.  Why do we need to link things into /usr/local/bin but we already ran python3 before that and seemed to assume it's the "right" python3?

Also, is this step only needed to do pycontroller?  Or will other builds and/or bootstrap also fail without this newer Python version?

Reply:
In #33979 is an update of all CI, pigweed and python packages

Reply:
I inserted these commands in the wrong order. My goal was to change the default version of python used in the shell to python3.11. Now I think that doing it this way will change the settings not only for our project but for the whole system. How should I pass information to use a specific version of Python during bootstrapping?

Reply:
I don't know how to tell bootstrap to use a specific Python, short of messing with PATH....

Note that installing via brew will likely end up using that Python for all sorts of things, since the brew stuff will come first in PATH.

Reply:
Hello @bzbarsky-apple, @jlatusek 
are we blocked here? any change we can continue with the upgrade of the stack to python 3.11 ?

Reply:
It needs to be done in sync with the upgrade to pigweed, because the version of pigweed we are using right now cannot bootstrap on an ARM Mac with python 3.11; see conversation above.